judgments/resolvers/document_resolver_engine.py

```diff
@@ -0,0 +1,20 @@
+from django.http.request import HttpRequest
+from django.views.generic import View
+
+from judgments.views.detail import detail, detail_xml, get_best_pdf, get_generated_pdf
+
+
+class DocumentResolverEngine(View):
+    def dispatch(self, request: HttpRequest, document_uri: str):
+        response_lookup = {
+            "/data.pdf": get_best_pdf,
+            "/generated.pdf": get_generated_pdf,
+            "/data.xml": detail_xml,
+            "/data.html": detail,
+        }
+        for key in response_lookup:
+            if document_uri.endswith(key):
```

Comment:
I feel like this significantly distorts the meaning of document_uri which is typically meaning "something of the form uksc/2024/1 or uksc/2024/1/press-summary/1, which has a direct correspondance to marklogic. I think I'd like to separate out the filename and path components..

Reply:
I think you're right here - I think it was a bit of naivety on my part coming to the project cold. I'll give it another go. Think I've got some ideas brewing 👍